apis/flowcollector/v1beta1/flowcollector_types.go

```diff
@@ -189,72 +189,81 @@ type EBPFMetrics struct {
 	DisableAlerts []EBPFAgentAlert `json:"disableAlerts"`
 }
 
-// `EBPFFlowFilter` defines the desired eBPF agent configuration regarding flow filtering
-type EBPFFlowFilter struct {
-	// Set `enable` to `true` to enable eBPF flow filtering feature.
-	Enable *bool `json:"enable,omitempty"`
-
-	// CIDR defines the IP CIDR to filter flows by.
-	// Example: 10.10.10.0/24 or 100:100:100:100::/64
+// `EBPFFlowFilterRule` defines the desired eBPF agent configuration regarding flow filtering rule.
```

Comment:
to avoid a breaking change, we should keep the EBPFFlowFilter as it is today. We can add new fields but not rename/remove existing fields.
So perhaps what I could suggest is to keep all the existing fields, and on top of that add the new `rules` field like you did. This is going to be ugly and will require at some point a new version that cleans it all.

Reply:
We can try. We need to check various places: the OLM console form, the generated asciidoc and also if an existing old FlowCollector wouldn't break.

Reply:
configure flowfilter old way works with latest agent changes
```yaml
flowFilter:
        action: Accept
        cidr: 2.2.2.2/24
        destPorts: 0
        enable: true
        ports: 0
        protocol: TCP
        sourcePorts: 53
        tcpFlags: SYN

```
```sh
oc get pods  -n netobserv-privileged netobserv-ebpf-agent-fpf9k -o yaml | grep "FLOW_FILTER_RULES" -A1
    - name: FLOW_FILTER_RULES
      value: '[{"ip_cidr":"2.2.2.2/24","protocol":"TCP","source_port":53,"action":"Accept","tcp_flags":"SYN"}]'
```